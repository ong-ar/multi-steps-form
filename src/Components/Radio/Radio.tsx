import * as React from "react";

interface IProps {
  name?: string;
  value?: string;
  disabled?: boolean;
  checked?: boolean;
  defaultChecked?: boolean;
  onChange?: (e: React.ChangeEvent<HTMLInputElement>) => void;
}

const Radio: React.SFC<IProps> = ({
  name,
  value,
  disabled,
  checked,
  defaultChecked,
  onChange,
  children
}) => {
  const inputStyle: React.CSSProperties = {
    marginLeft: -20,
    position: "absolute"
  };

  const labelStyle: React.CSSProperties = {
    cursor: "pointer",
    paddingLeft: 20,
    paddingRight: 10
  };
  return (
    <div>
      <label style={labelStyle}>
        <input
          type="radio"
          name={name}
          value={value}
          disabled={disabled}
          checked={checked}
          defaultChecked={defaultChecked}
          onChange={onChange}
          style={inputStyle}
        />
        {children}
      </label>
    </div>
  );
};

export default Radio;
