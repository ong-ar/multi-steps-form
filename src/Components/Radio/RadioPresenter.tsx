import * as React from "react";

interface IProps {
  name?: string;
  value?: string;
  disabled?: boolean;
  checked?: boolean;
  onChange?: (e: React.ChangeEvent<HTMLInputElement>) => void;
}

const RadioPresenter: React.SFC<IProps> = ({
  name,
  value,
  disabled,
  checked,
  onChange,
  children
}) => {
  const inputStyle: React.CSSProperties = {
    marginLeft: -20,
    position: "absolute"
  };

  const labelStyle: React.CSSProperties = {
    cursor: "pointer",
    paddingLeft: 20,
    paddingRight: 10
  };
  return (
    <label style={labelStyle}>
      <input
        type="radio"
        name={name}
        value={value}
        disabled={disabled}
        checked={checked}
        onChange={onChange}
        style={inputStyle}
      />
      {children}
    </label>
  );
};

export default RadioPresenter;
