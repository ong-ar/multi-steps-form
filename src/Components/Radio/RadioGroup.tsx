import * as React from "react";
import Radio from "./Radio";

interface IProps {
  name: string;
  defaultValue?: string;
  onChange?: (checkedValue: string, groupId?: string | number) => void;
  groupId?: string | number; // 부모노드에서 ID가 필요할 경우 사용
}

const RadioGroup: React.FC<IProps> = ({
  name,
  defaultValue,
  onChange,
  children,
  groupId
}) => {
  // Radio Change handler
  const onRadioChange: React.ChangeEventHandler<HTMLInputElement> = e => {
    // RadioGroup onChange
    if (onChange) {
      onChange(e.target.value, groupId);
    }
  };

  const radioGroup = {
    name,
    onChange: onRadioChange
  };

  const customedChildren = React.Children.map(
    children,
    (child: React.ReactElement) => {
      if (child && child.type === Radio) {
        // defaultValues 적용
        if (defaultValue === child.props.value) {
          return React.cloneElement(child, {
            ...radioGroup,
            defaultChecked: true
          });
        } else {
          return React.cloneElement(child, {
            ...radioGroup,
            defaultChecked: false
          });
        }
      } else {
        return child;
      }
    }
  );
  return <div>{customedChildren}</div>;
};

export default RadioGroup;
