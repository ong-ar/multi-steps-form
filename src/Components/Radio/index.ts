import RadioPresenter from "./RadioPresenter";
export default RadioPresenter;
