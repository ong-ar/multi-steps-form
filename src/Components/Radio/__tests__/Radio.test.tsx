import * as React from "react";
import { shallow } from "enzyme";

import Radio from "../Radio";

// radio
it("Radio", () => {
  const wrapper = shallow(<Radio />);
  expect(wrapper).toMatchSnapshot();
});

// props: name 체크
it("Radio:name", () => {
  const wrapper = shallow(<Radio name="radio" />);
  expect(wrapper.find("input").prop("name")).toEqual("radio");
});

// props: value 체크
it("Radio:value", () => {
  const wrapper = shallow(<Radio value="radio" />);
  expect(wrapper.find("input").prop("value")).toEqual("radio");
});

// props: disabled 체크
it("Radio:disabled", () => {
  const wrapper = shallow(<Radio disabled={true} />);
  expect(wrapper.find("input").prop("disabled")).toEqual(true);
});

// props: checked 체크
it("Radio:checked", () => {
  const wrapper = shallow(<Radio checked={false} />);
  expect(wrapper.find("input").prop("checked")).toEqual(false);
});

// props: defaultChecked 체크
it("Radio:defaultChecked", () => {
  const wrapper = shallow(<Radio defaultChecked={true} />);
  expect(wrapper.find("input").prop("defaultChecked")).toEqual(true);
});
