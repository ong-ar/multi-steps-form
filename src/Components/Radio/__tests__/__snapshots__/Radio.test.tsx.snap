// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Radio 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Radio />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <input
          checked={undefined}
          defaultChecked={undefined}
          disabled={undefined}
          name={undefined}
          onChange={undefined}
          style={
                    Object {
                              "marginLeft": -20,
                              "position": "absolute",
                            }
          }
          type="radio"
          value={undefined}
/>,
        undefined,
      ],
      "style": Object {
        "cursor": "pointer",
        "paddingLeft": 20,
        "paddingRight": 10,
      },
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "checked": undefined,
          "defaultChecked": undefined,
          "disabled": undefined,
          "name": undefined,
          "onChange": undefined,
          "style": Object {
            "marginLeft": -20,
            "position": "absolute",
          },
          "type": "radio",
          "value": undefined,
        },
        "ref": null,
        "rendered": null,
        "type": "input",
      },
      undefined,
    ],
    "type": "label",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <input
            checked={undefined}
            defaultChecked={undefined}
            disabled={undefined}
            name={undefined}
            onChange={undefined}
            style={
                        Object {
                                    "marginLeft": -20,
                                    "position": "absolute",
                                  }
            }
            type="radio"
            value={undefined}
/>,
          undefined,
        ],
        "style": Object {
          "cursor": "pointer",
          "paddingLeft": 20,
          "paddingRight": 10,
        },
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "checked": undefined,
            "defaultChecked": undefined,
            "disabled": undefined,
            "name": undefined,
            "onChange": undefined,
            "style": Object {
              "marginLeft": -20,
              "position": "absolute",
            },
            "type": "radio",
            "value": undefined,
          },
          "ref": null,
          "rendered": null,
          "type": "input",
        },
        undefined,
      ],
      "type": "label",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
